infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -339,11 +339,27 @@ private synchronized void runDetectionTasks() {
             // Corrects out of phase epoch issues if present in the report. This method
             // performs re-sealing of all nodes if required and catchup of a layout server to
             // the current state.
-            Layout updatedLayout = correctWrongEpochs(pollReport, layout);
+            final Layout updatedLayout = correctWrongEpochs(pollReport, layout);
 
             Result<DetectorTask, RuntimeException> failure = Result.of(() -> {
-                if (pollReport.isCurrentLayoutSlotUnFilled()) {
-                    detectFailure(updatedLayout, Collections.emptySet(), pollReport).join();
+
+                // This is just an optimization in case we receive a WrongClusterException
+                // while one of the other management clients is trying to move to a new layout.
+                // This check is merely trying to minimize the the scenario in which we end ip
+                // filling the slot with an outdated layout. Due to potential failures, we cannot
+                // wait for more than the majority of layout servers.
+                if (!pollReport.isMajorityLayoutServersSealed(updatedLayout)) {
+                    log.info(("Majority of the layout servers have not been sealed. Skipping.");
```

Comment:
maybe we can reduce it to trace? In the other case, we would have many many of this messages in the log

Reply:
Reduced it to debug, This message will be printed only during WrongEpochException, so it should not cause much chatter.